File: LC283.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int temp, k = 0;

        while(k < nums.size())
            if(nums[k] == 0)
                break;
            else
                k++;

        int i = k, j = k+1;

        while(i < nums.size() && j < nums.size()){
            if(nums[j] != 0){
                temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                i++;
            }
            j++;
        }

        return;
    }
};

File: LC283.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # move to first zero place
        k = 0
        while k < len(nums):
            if not nums[k]:
                break
            k += 1
        
        i = k
        j = k+1

        while i < len(nums) and j < len(nums):
            if nums[j]:
                temp = nums[i]
                nums[i] = nums[j]
                nums[j] = temp
                i += 1
            
            j += 1
        
        return
